framework/include/transfers/MultiAppNearestNodeTransfer.h

```diff
@@ -51,12 +48,31 @@ class MultiAppNearestNodeTransfer :
    */
   Node * getNearestNode(const Point & p, Real & distance, MooseMesh * mesh, bool local);
 
+  /**
+   * Return the distance between the given point and the farthest corner of the
+   * given bounding box.
+   * @param p The point to evaluate all distances from.
+   * @param bbox The bounding box to evaluate the distance to.
+   * @return The maximum distance between the point p and the eight corners of
+   * the bounding box bbox.
+   */
+  Real bboxMaxDistance(Point p, MeshTools::BoundingBox bbox);
+
+  /**
+   * Return the distance between the given point and the nearest corner of the
+   * given bounding box.
+   * @param p The point to evaluate all distances from.
+   * @param bbox The bounding box to evaluate the distance to.
+   * @return The minimum distance between the point p and the eight corners of
+   * the bounding box bbox.
+   */
+  Real bboxMinDistance(Point p, MeshTools::BoundingBox bbox);
+
+  std::vector<Node *> getLocalNodes(MooseMesh * mesh);
```

Comment:
This returns a copy of a vector and you call it in a loop... It may be more efficient to have this function take a reference to a `std::vector<Node*>` provided by the user.  This will cut down on some vector allocations/deallocations.